hal/src/main/native/athena/PowerDistribution.cpp

```diff
@@ -15,18 +16,24 @@ using namespace hal;
 extern "C" {
 
 HAL_PowerDistributionHandle HAL_InitializePowerDistribution(
-    int32_t moduleNumber, HAL_PowerDistributionType type, int32_t* status) {
+    int32_t moduleNumber, HAL_PowerDistributionType type,
+    const char* allocationLocation, int32_t* status) {
   if (type == HAL_PowerDistributionType::HAL_PowerDistributionType_kAutomatic) {
     type = HAL_PowerDistributionType::HAL_PowerDistributionType_kCTRE;
   }
```

Comment:
Is there any plan to try to be able to do actual auto-detection here? The "automatic" type is exposed in the high-level APIs, and that feels very confusing if it's just going to default to CTRE. If we can't do actual auto-detection, it seems like we should remove the `kAutomatic` value from the relevant enums, and pass `kCTRE` from the param-less `PowerDistribution` constructors in the high-level APIs for backwards compatibility.

Reply:
Yes, there is going to be a way to do automatic detection. We just don't have it yet.

Reply:
Awesome!